### ISSUE TITLE
Cancel Sync

### ISSUE THREAD

**Issue by mcichowski** [airbytehq/airbyte#1623]
## Tell us about the problem you're trying to solve
If e.g. a Salesforce to Postgres sync fails due to connection limits there is no way to abort the sync from the UI.
The sync job keeps retrying

## Describe the solution you’d like
There should be a simple cancel action

## Describe the alternative you’ve considered or used
Restart the whole thing

**Comment by michel-tricot** [comment]
Thank you @mcichowski! Will look at it

**Comment by mcichowski** [comment]
I've seen there is a hard retry limit of 5 in the JobRetrier this should be ideally configurable

**Comment by jrhizor** [comment]
Ideally we would be able to cancel any running job (sync or otherwise).

Syncing would probably need custom UI work so it's clear how to cancel it, but other jobs could show in the admin page (maybe?) and be killed there.

**Comment by cgardens** [comment]
Couple things:
* 5 is a lot. I don't know what we picked it (and likely it was I who picked it), but 3 might just be better. I doubt main jobs fail on their first 3 runs and then succeed on their 4th or 5th. 🤷‍♀️ 
* Adding cancellation is on the list of things I don't want to do until we replace the scheduler with a "real" workflow engine (e.g. airflow etc). This is because a workflow engine will have this feature built in, so anything we build now in the current scheduler will be throwaway work.

**Comment by roshan** [comment]
Interesting failure mode for me, today. I added a new table from my source Postgres into my sync, it failed because I didn't have permission on that table (fine, my problem), but now I'm sitting there waiting for the retries to complete before I can do anything, which takes forever.

Honestly, I'm not even sure what the retry functionality is getting me. I have a 15 min sync frequency, going incremental. If it doesn't work in this attempt, it'll work in the next.

For the moment, I'm just going to drop attempts down in my fork to 1.

I made the retries issue separately [here](https://github.com/airbytehq/airbyte/issues/2493)
